grabbit/src/test/groovy/com/twcable/grabbit/jcr/ProtoNodeDecoratorSpec.groovy

```diff
@@ -0,0 +1,127 @@
+package com.twcable.grabbit.jcr
+
+import com.twcable.grabbit.proto.NodeProtos
+import com.twcable.grabbit.proto.NodeProtos.Node as ProtoNode
+import spock.lang.Specification
+
+import javax.jcr.Node
+import javax.jcr.Session
+
+import static javax.jcr.PropertyType.STRING
+import static org.apache.jackrabbit.JcrConstants.JCR_MIXINTYPES
+import static org.apache.jackrabbit.JcrConstants.JCR_PRIMARYTYPE
+
+@SuppressWarnings("GroovyAccessibility")
+class ProtoNodeDecoratorSpec extends Specification {
+
+    ProtoNode decoratedProtoNode
+
+
+    def setup() {
```

Comment:
Here you have used `setup()` method that `run before every feature method` do you need to set before every feature method or this is just one time setup? 

Also note that `ProtoNodeDecorator can not be constructed with a null ProtoNode` feature is not using the `setup()` values but the `setup()` will fun for that feature too.